src/trezor-keyring.ts

```diff
@@ -234,35 +293,46 @@ class TrezorKeyring extends EventEmitter {
    * Accepts either an ethereumjs-tx or @ethereumjs/tx transaction, and returns
    * the same type.
    *
-   * @template {TypedTransaction | OldEthJsTransaction} Transaction
-   * @param {string} address - Hex string address.
-   * @param {Transaction} tx - Instance of either new-style or old-style ethereumjs transaction.
-   * @returns {Promise<Transaction>} The signed transaction, an instance of either new-style or old-style
+   * @param address - Hex string address.
+   * @param tx - Instance of either new-style or old-style ethereumjs transaction.
+   * @returns The signed transaction, an instance of either new-style or old-style
    * ethereumjs transaction.
    */
-  signTransaction(address, tx) {
+  async signTransaction(
+    address: string,
+    tx: TypedTransaction | OldEthJsTransaction,
+  ) {
     if (isOldStyleEthereumjsTx(tx)) {
       // In this version of ethereumjs-tx we must add the chainId in hex format
       // to the initial v value. The chainId must be included in the serialized
       // transaction which is only communicated to ethereumjs-tx in this
       // value. In newer versions the chainId is communicated via the 'Common'
       // object.
-      return this._signTransaction(address, tx.getChainId(), tx, (payload) => {
-        tx.v = Buffer.from(payload.v, 'hex');
-        tx.r = Buffer.from(payload.r, 'hex');
-        tx.s = Buffer.from(payload.s, 'hex');
-        return tx;
-      });
+      return this.#signTransaction(
+        address,
+        // @types/ethereumjs-tx and old ethereumjs-tx versions document
+        // this function return value as Buffer, but the actual
+        // Transaction._chainId will always be a number.
+        // See https://github.com/ethereumjs/ethereumjs-tx/blob/v1.3.7/index.js#L126
+        tx.getChainId() as unknown as number,
```

Comment:
We can remove `as unknown as number` and let the type pass through as a `Buffer` - see overload comment.